Feature/web diary

gochicus opened the pull request.

gochicus committed:
chore: environment settings

ISSUES CLOSED: #62

gochicus committed:
feat(web-diary): app construction files added

affects: @app-launcher/app

ISSUES CLOSED: #62

gochicus committed:
feat(web-diary): app components added

affects: @ui/app-header, @ui/button, @ui/input, @ui/layout, @ui/post-add-form, @ui/post-list-item,
@ui/post-list, @ui/filter, @ui/search

ISSUES CLOSED: $62

gochicus committed:
feat(web-diary): elements for style added

affects: @app-launcher/app, @ui/button, @ui/post-add-form, @ui/post-list-item, @ui/post-list,
@ui/filter

Box elements with flex-basis, flex options added to elements, styled elements added

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/package.json:
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "web-diary",
```
@practice?

TFK70 commented on practice/web-diary/package.json:
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "web-diary",
+  "version": "1.0.0",
```
версии должны начинаться с 0.0.0

TFK70 commented on practice/web-diary/package.json:
```diff
@@ -0,0 +1,33 @@
+  },
+  "dependencies": {
+    "@app-launcher/app": "0.0.0",
+    "@types/react": "^17.0.3",
```
Типы - дев зависимости
От них нет никакого смысла при установке пакета как зависимость т.к. типизация нужна только при разработке

TFK70 commented on practice/web-diary/ui/layout/src/Box.tsx:
```diff
@@ -0,0 +1,56 @@
+  flexbox?: string
+  boxShadow?: string
+  boxSizing?: string
+  position?: string
```
чтобы не описывать кучу пропсов руками стоит воспользоваться [styled-system](https://styled-system.com/api)
увидел что он тут есть, тогда воспользуйся [типами](https://github.com/Atlantis-Lab/uikit/blob/master/packages/select/src/types.ts#L2) оттуда

TFK70 replied:
да и в целом было бы неплохо держать интерфейсы в отдельном файле, как в той ссылке

TFK70 commented on practice/web-diary/ui/post-add-form/src/index.tsx:
```diff
@@ -0,0 +1,3 @@
+import PostAddForm from './PostAddForm'
+
+export default PostAddForm
```
в индексе не должно быть дефолтных экспортов, сделай так же как ты сделал в layout
и в целом нужно стараться писать в одном стиле

TFK70 replied:
кстати по смежной причине появляются ворнинги при сборке проекта

TFK70 commented on practice/web-diary/app-launcher/app/src/App.tsx:
```diff
@@ -0,0 +1,39 @@
+const App = () => {
+  return (
+    <Background>
+      <Layout display='flex' flexDirection='column' alignItems='center' margin='0' maxWidth='800'>
```
display: flex уже есть в дефолт пропсах лейаута

TFK70 commented on practice/web-diary/app-launcher/app/package.json:
```diff
@@ -0,0 +1,22 @@
+  "license": "BSD 3-Clause",
+  "main": "src/index.tsx",
+  "scripts": {
+    "lint": "eslint"
```
должно быть в корне + тут все-таки должен быть actl

TFK70 commented on practice/web-diary/ui/app-header/package.json:
```diff
@@ -0,0 +1,14 @@
+  "license": "BSD 3-Clause",
+  "main": "src/index.tsx",
+  "scripts": {
+    "lint": "eslint"
```
ну соответственно подобное надо бы убрать везде
линт достаточно запустить 1 раз из корня, ему незачем в скриптах в каждом пакете висеть

TFK70 commented on practice/web-diary/ui/button/src/ButtonInfo.tsx:
```diff
@@ -0,0 +1,9 @@
+
+export const ButtonInfo = () => {
+  const intl = useIntl()
+  return <button type='button'>{intl.formatMessage(messages.info)}</button>
```
ни onClick не повесить, ни стилизовать
компонент должен быть гибким и конфигурируемым (в пределах разумного, конечно)

TFK70 commented on practice/web-diary/ui/button/src/ButtonSubmit.tsx:
```diff
@@ -0,0 +1,9 @@
+
+export const ButtonSubmit = () => {
+  const intl = useIntl()
+  return <button type='submit'>{intl.formatMessage(messages.add)}</button>
```
Все кнопки отличаются лишь контентом внутри? Не было смысла разбивать их на много разных компонентов, в таком случае
Конструкция
```javascript
const Button = ({ children }) => <button>{children}</button>
```
Способна сгрести все эти компоненты в один универсальный
И да, опять же, children - не все, что должно быть в ее пропсах
Должно быть +-[так](https://github.com/Atlantis-Lab/uikit/blob/master/packages/button/src/Button.tsx#L33)

TFK70 replied:
Это кстати в том числе про второй принцип SOLID: открытости/закрытости
Руководствуйся им при проектировании

TFK70 commented on practice/web-diary/ui/input/src/Input.tsx:
```diff
@@ -0,0 +1,23 @@
+`
+
+export const Input = () => {
+  return <StyledInput type='text' />
```
И снова то же самое
Вот понадобится тебе сделать
```javascript
background-color: '#000',
color: '#fff'
```
Неужели нужно будет создавать новый компонент для этого? Его нужно открыть как минимум для модификации

TFK70 commented on practice/web-diary/ui/post-list/src/PostList.tsx:
```diff
@@ -0,0 +1,22 @@
+    <List>
+      <PostListItem />
+      <PostListItem />
+      <PostListItem />
```
А если их нужно будет 4? 50? 100?
Снова очень не гибкий компонент, так не должно быть

TFK70 commented on practice/web-diary/ui/post-status-filter/src/PostStatusFilter.tsx:
```diff
@@ -0,0 +1,25 @@
+    <Filter>
+      <Box display='flex' justifyContent='space-between'>
+        <Box>
+          <ButtonInfo />{' '}
```
Хардкод
Если есть проблема с пробелами - ее нужно решить в другом месте, другим способом, в лучшем случае - автоматизировать, в крайнем - вынести эту логику в сам баттон. Адаптируй свои компоненты к окружению

TFK70 commented on practice/web-diary/ui/search-panel/src/SearchPanel.tsx:
```diff
@@ -0,0 +1,9 @@
+import { Input } from '@ui/input'
+
+const SearchPanel = () => {
+  return <Input />
```
Какой смысл использовать SearchPanel если он абсолютно идентичен инпуту?

gochicus committed:
refactor(web-diary): requested changes made

affects: @app-launcher/app, @practice/web-diary, @ui/button, @ui/input, @ui/layout,
@ui/post-add-form, @ui/post-list-item, @ui/post-list, @ui/filter

components refactored, package.json files changed

ISSUES CLOSED: #62

gochicus committed:
chore: webpack cinfig changed

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/app-launcher/app/src/App.tsx:
```diff
@@ -0,0 +1,45 @@
+import { PostAddForm }      from '@ui/post-add-form'
+import { PostList }         from '@ui/post-list'
+
+import { Input }            from '../../../ui/input'
```
Релятивные пути
Приемущество `'@ui/input` над этим кошмаром очевидно

TFK70 commented on practice/web-diary/ui/button/src/CommonButton.tsx:
```diff
@@ -0,0 +1,10 @@
+import styled                                      from '@emotion/styled'
+import { border, color, flexbox, layout, padding } from 'styled-system'
+
+import { ButtonInterface }                         from '../interface/interface'
```
Типы можно вынести просто в файл types.ts
Ну а при таком подходе папку стоило назвать interfaceS, а файлу с интерфейсом для кнопки более конкретное название

TFK70 commented on practice/web-diary/ui/input/src/Input.tsx:
```diff
@@ -0,0 +1,23 @@
+import styled from '@emotion/styled'
+
+const StyledInput = styled.input`
+  display: block;
```
https://github.com/Atlantis-Academy/frontend-study-ikorol/pull/66#discussion_r603411408

TFK70 commented on practice/web-diary/ui/input/src/Input.tsx:
```diff
@@ -0,0 +1,23 @@
+  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
+`
+
+export const Input = ({ text }, { color }) => {
```
({ text, color }) выглядит приятнее

gochicus replied:
Так отчего-то не работает.

TFK70 commented on practice/web-diary/ui/post-list/src/PostList.tsx:
```diff
@@ -0,0 +1,14 @@
+import styled from '@emotion/styled'
+
+const List = styled.ul(
+  ' margin-top: 50px;\n' +
```
Выглядит очень плохо,ничто не мешало стилизовать это так же, как и другие компоненты
И реализовать подобное можно без ul

TFK70 commented on practice/web-diary/ui/post-list-item/src/PostListItem.tsx:
```diff
@@ -0,0 +1,38 @@
+  '  transition: 0.5s all;',
+)
+
+const ListItem = styled.li('font-size: 1.25 rem \n', 'display: flex \n', 'justify-content: row')
```
Соотвтственно, так писать не стоит

TFK70 commented on practice/web-diary/ui/post-status-filter/src/PostStatusFilter.tsx:
```diff
@@ -0,0 +1,24 @@
+import { CommonButton } from '@ui/button'
+import { Box }          from '@ui/layout'
+
+import messages         from '../../post-add-form/messages/messages'
```
Полагаю из-за невнимательности
Путь эквивалентен `../messages/messages`

TFK70 commented on practice/web-diary/package.json:
```diff
@@ -0,0 +1,34 @@
+    "@typescript-eslint/parser": "^4.20.0"
+  },
+  "dependencies": {
+    "@app-launcher/app": "0.0.0",
```
Он тут зачем?

TFK70 replied:
И я все еще не понимаю зачем он здесь нужен

TFK70 commented on practice/web-diary/ui/button/interface/interface.tsx:
```diff
@@ -0,0 +1,4 @@
+export interface ButtonInterface {
+  color?: string
```
И да, я все еще настаиваю на использовании типов из styled-system)

gochicus replied:
@TFK70 Я пока не готов пользоваться этой автоматизацией, в интерфейсе кнопки прописал ColorProps - получилось, в лэйауте применение этих пропсов вызывает ошибки тайпскрипта, с чем они связаны я даже предположить не берусь. 

UPD. Ещё интересней, то что работало с ColorProps ( цвета применялись) вдруг перестало работать, не смог сделать коммит, так как вновь появилась ошибка тайпскрипт на ColorProps

TFK70 replied:
> @TFK70 Я пока не готов пользоваться этой автоматизацией, в интерфейсе кнопки прописал ColorProps - получилось, в лэйауте применение этих пропсов вызывает ошибки тайпскрипта, с чем они связаны я даже предположить не берусь.
> 
> UPD. Ещё интересней, то что работало с ColorProps ( цвета применялись) вдруг перестало работать, не смог сделать коммит, так как вновь появилась ошибка тайпскрипт на ColorProps

Опиши более подробно, приложи логи, свои действия

gochicus committed:
refactor(web-diary): styles changed, paths changed

affects: @app-launcher/app, @practice/web-diary, @ui/button, @ui/input, @ui/post-add-form,
@ui/post-list-item, @ui/post-list, @ui/filter

ISSUES CLOSED: #62

gochicus committed:
refactor(web-diary): prop added in component

affects: @ui/button, @ui/post-list-item

ISSUES CLOSED: #62

gochicus committed:
refactor(web-diary): refactor according eslint rules

affects: @ui/button, @ui/input, @ui/post-add-form, @ui/post-list-item, @ui/filter

ISSUES CLOSED: #62

gochicus committed:
style(web-diary): prettier ccde fix

affects: @ui/button

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/ui/button/src/CommonButton.tsx:
```diff
@@ -0,0 +1,7 @@
+
+import { Button } from '@ui/button'
+
+export const CommonButton = ({ children, color }) => {
```
Не могу понять к чему этот компонент
Разве я не могу использовать просто Button?
```javascript
<Button color='aqua'>{'OK'}</Button>
```

gochicus replied:
@TFK70 Не совсем понял, Button просто стилизованная кнопка, а вынес в отдельный файл потому что eslint выдает ошибку из-за пропса color и color в Button.

TFK70 commented on practice/web-diary/ui/input/src/Input.tsx:
```diff
@@ -0,0 +1,19 @@
+import React  from 'react'
+import styled from '@emotion/styled'
+
+const StyledInput = styled('input')(() => ({
```
Компонент по-прежнему очень скудный
Дай возможность его конфигурации снаружи: поменять цвет, размер
Оставь только базовые пропсы которые уж точно нигде менять не нужно будет

TFK70 commented on practice/web-diary/ui/input/src/Input.tsx:
```diff
@@ -0,0 +1,19 @@
+  borderRadius: '3px',
+}))
+
+export const Input = ({ text }) => {
```
Проп для placeholder лучше и назвать placeholder

TFK70 commented on practice/web-diary/ui/layout/interfaces/LayoutInterface.ts:
```diff
@@ -0,0 +1,38 @@
+import { system } from 'styled-system'
+
+export interface Row {
```
Тогда к чему было выносить все типы в отдельную директорию если все интерфейсы все-равно смешаны в одном файле? Проще уж было на том же уровне файл types.ts создать и скинуть все туда - тогда это можно понять
При таком подходе интерфейсы нужно декомпозировать. Вынеси каждый интерфейс в свой файл, так же намного удобнее будет ориентироваться

TFK70 replied:
НО, так же создай index и экспортируй все интерфейсы через него
Это необходимо для того чтобы избежать дублирования:
```javascript
import { LayoutInterface } from './interfaces/LayoutInterface'
import { BoxInterface } from './interfaces/BoxInterface'
```
С индекс файликом будет так:
```javascript
import { LayoutInterface,BoxInteface } from './interfaces'
```

TFK70 commented on practice/web-diary/ui/layout/interfaces/LayoutInterface.ts:
```diff
@@ -0,0 +1,38 @@
+  justifyContent?: string
+  alignItems?: string
+}
+export const boxSizing = system({
```
И зачем он здесь?

TFK70 commented on practice/web-diary/ui/layout/src/index.tsx:
```diff
@@ -0,0 +1,5 @@
+import { Box }    from './Box'
```
Если ты не используешь дефолтные экспорты - в этой конструкции нет необходимости
Можно просто сделать так:
```javascript
export * from './Layout'
```

gochicus replied:
@TFK70  Этот момент тоже непонятен, я ж использую именованные экспорты.  Зачем экспортировать всё чеерз звездочку?

TFK70 replied:
> @TFK70 Этот момент тоже непонятен, я ж использую именованные экспорты. Зачем экспортировать всё чеерз звездочку?

Звездочка абсолютно идентична текущей конструкции, она только сократит код в 2 раза

TFK70 commented on practice/web-diary/ui/post-list/src/PostList.tsx:
```diff
@@ -0,0 +1,14 @@
+import styled from '@emotion/styled'
+
+const List = styled.div`
+  margin-top: 10px;
```
Это тоже нужно описать js-объектом, как я раньше уже упоминал

gochicus committed:
refactor(web-diary): styled elements refactored

affects: @app-launcher/app, @ui/button, @ui/input, @ui/layout, @ui/post-add-form, @ui/post-list

styled elements refactored, components refactored, exports refactored

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/ui/input/src/Input.tsx:
```diff
@@ -1,19 +1,33 @@
+import { border, borderRadius, system } from 'styled-system'
 
-const StyledInput = styled('input')(() => ({
+import { InputCSS }                     from './Types'
```
Почему файл с заглавной буквы-то?
И что значит InputCSS? Ты описываешь типы для пропсов инпута, а не для css

gochicus committed:
WIP(web-diary): error test

affects: @app-launcher/app, @ui/button, @ui/input, @ui/post-add-form

ISSUES CLOSED: #62

gochicus committed:
refactor(web-diary): components refactored

affects: @ui/button, @ui/input, @ui/post-add-form, @ui/post-list-item

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/app-launcher/app/src/App.tsx:
```diff
@@ -0,0 +1,40 @@
+          <PostStatusFilter />
+        </Layout>
+        <PostList>
+          <PostListItem />
```
Насчет этого я уже писал, если не ошибаюсь
Эта логика должна быть вынесена внутрь PostList. Создание/конфигурация итемов должна происходить через изменение пропсов PostList

TFK70 commented on practice/web-diary/ui/app-header/src/AppHeader.tsx:
```diff
@@ -0,0 +1,17 @@
+  const intl = useIntl()
+  return (
+    <Box justifyContent='space-between' alignItems='flex-end' display='flex'>
+      <h2>{intl.formatMessage(messages.header)}</h2>
```
Здесь h2, в другом месте h1, в третьем месте h3. Создай для текста отдельный компонент Text и производи стилизацию текста через пропсы. Так намного проще сохранить единый стиль кода и зарегламентировать все возможные стилизации текста

TFK70 commented on practice/web-diary/ui/post-add-form/src/PostAddForm.tsx:
```diff
@@ -0,0 +1,33 @@
+        </Box>
+        <Box flexBasis='40px' />
+        <Box>
+          <CommonButton color='black'>{intl.formatMessage(messages.add)}</CommonButton>
```
От CommonButton тоже нужно избавиться, он ничем не отличается от обычного Button

gochicus replied:
@TFK70 Я не знаю как props children передать в Button 
```

const Button = ({ children, ...props }) => (
  <StyledButton {...props}>{divideChildren(children, ContentPart)}</StyledButton>
)
```
Это из ukit, логика та же, в компонент передается стилизованная кнопка, а ужев ней размещены чилдрен как пропсы самого компонента.  Я в данном случае должен что сделать?

TFK70 replied:
> @TFK70 Я не знаю как props children передать в Button

В смысле?
```javascript
<Button color='black'>{intl.formatMessage(messages.add)}</Button>
```
Достаточно везде CommonButton поменять на Button

TFK70 replied:
@gochicus в uikit делается прослойка в виде еще одного компонента:
```javascript
const StyledButton = styled('button')((styles) => {...styles})

const Button = ({children, ...props}) => <StyledButton {...props}>{children}</Button>

export default Button
```
В твоем случае кнопка экспортируется напрямую:
```javascript
const StyledButton = styled('button')((styles) => {...styles})

export default StyledButton
```
Только вместо StyledButton она называется Button
Тут вообще никаких проблем нет

gochicus replied:
Ок, понял.

TFK70 commented on practice/web-diary/ui/post-list/src/PostList.tsx:
```diff
@@ -0,0 +1,14 @@
+}))
+
+const PostList = ({ children }) => {
+  return <List>{children}</List>
```
Как раз здесь и нужно описать логику создания итемов, а не делать это снаружи

gochicus committed:
refactor(web-diary): components refactored and deleted

affects: @app-launcher/app, @ui/app-header, @ui/button, @ui/input, @ui/post-list, @ui/text

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/ui/button/src/Button.tsx:
```diff
@@ -1,15 +1,9 @@
+import { border, color, layout, padding } from 'styled-system'
 
-import { ButtonInterface }                         from './types'
+import { ButtonInterface }                from './types'
```
ButtonProps
Ни к чему приписывать Interface, это и так понятно что в файле types кроме интерфейсов ничего быть не может, но вот подписать что именно он описывает - не помешает

TFK70 commented on practice/web-diary/ui/input/src/Input.tsx:
```diff
@@ -1,7 +1,14 @@
 
-export const Input = styled<any>('input')(border, layout, color, typography, space, display)
+export const Input = styled<any>('input')(
+  { border },
```
Это как?

TFK70 commented on practice/web-diary/ui/post-list/src/PostList.tsx:
```diff
@@ -1,14 +1,52 @@
+  transition: ' 0.5s all',
+}))
+
+const diary: { post: string; id: number }[] = [
```
diary должен приходить извне
В данном случае зона ответственности твоего компонента размывается + он не поддается конфигурации
Твоя задача - описать шаблон компонента. Представь что помимо тебя этим компонентом будут пользоваться другие разрабы для верстки своих страниц, а там контекст совсем другой

TFK70 commented on practice/web-diary/ui/post-list/src/PostList.tsx:
```diff
@@ -1,14 +1,52 @@
+  const elements = diary.map((item: any) => {
+    return (
+      <List key={item.id}>
+        <Layout justifyContent='space-around'>
```
Зачем это описывать вручную?
У тебя же был компонент PostListItem.  Можно было поставить его сюда как зависимость, ничего плохого в этом нет, скорее наоборот

gochicus committed:
refactor(web-diary): components refactored

affects: @ui/button, @ui/input, @ui/post-list-item, @ui/post-list

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/ui/button/src/Button.tsx:
```diff
@@ -1,9 +1,9 @@
 import { border, color, layout, padding } from 'styled-system'
 
-import { ButtonInterface }                from './types'
+import { ButtonChildrenAndColor }         from './types'
```
Ну это слишком конкретно
Если тебе понадобится добавить в пропсы border - придется переименовываться в ButtonChildrenAndColorWithBorder?)
Просто BorderProps более чем достаточно

TFK70 commented on practice/web-diary/ui/post-list/src/PostList.tsx:
```diff
@@ -1,48 +1,20 @@
 
-import { Button }      from '../../button'
-import { Box, Layout } from '../../layout'
+import { PostListItem } from '../../post-list-item'
```
Снова релятивные импорты?

TFK70 commented on practice/web-diary/ui/post-list/src/PostList.tsx:
```diff
@@ -1,48 +1,20 @@
-import { Button }      from '../../button'
-import { Box, Layout } from '../../layout'
+import { PostListItem } from '../../post-list-item'
+import { Diary }        from './Diary'
```
Нет-нет-нет, я не это имел ввиду
Под "компонентом" подразумевается весь пакет `@ui/post-list`. Единственное за что он должен отвечать - за представление данных, которые ему предоставляют. В этом случае данные предоставляются через пропсы, твой компонент не должен ничего знать о том, что в него войдет, но должен с помощью типизации говорить о том,  в каком виде эти данные нужно предоставлять

gochicus replied:
@TFK70  Как то слабо себе представляю реализацию, можешь примерно обозначить где значение пропса быть должно?

TFK70 replied:
> @TFK70 Как то слабо себе представляю реализацию, можешь примерно обозначить где значение пропса быть должно?

В смыысле?
Пропсы для компонента можно определить только в одном месте, оттуда ты этот diary и возьмешь
Ты будешь его прокидывать как атрибут
```javascript
<PostList diary={{ name: 'name1`, id: 1 }} />
```

gochicus replied:
@TFK70 Вот есть у меня пропс в  PostListItem,  он получал значение пропса в PostList, а если  в  PostList просто пропс, откуда должен получать значение PostListItem?

TFK70 replied:
> @TFK70 Вот есть у меня пропс в PostListItem, он получал значение пропсав PostList, а если в PostList просто пропс, откуда должен получать значение PostListItem?

Снова ничего не понял, но предполагаю что ты про пробрасывание пропсов вниз по дереву, если да - то это нормально
Думаю будет странно если я тебе готовый кусок кода напишу, тут вполне можно самому все сделать если немного внимательнее быть

Пропсы - это всего лишь аргумент функции который по дефолту объект. Ты можешь делать с ним все, что угодно

gochicus committed:
refactor(web-diary): components refactored

affects: @app-launcher/app, @ui/button, @ui/post-list

ISSUES CLOSED: #62

gochicus committed:
refactor(web-diary): component refactored

affects: @app-launcher/app, @ui/post-list

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/app-launcher/app/src/App.tsx:
```diff
@@ -0,0 +1,41 @@
+import { PostAddForm }      from '@ui/post-add-form'
+import { PostList }         from '@ui/post-list'
+
+const Background = styled('div')(() => ({
```
А почему он здесь? Это отдельный компонент

TFK70 commented on practice/web-diary/ui/app-header/src/AppHeader.tsx:
```diff
@@ -0,0 +1,18 @@
+
+import messages    from '../messages/messages'
+
+const AppHeader = () => {
```
Теперь пора вынести подобные компоненты в components
Они слишком привязаны к контексту для ui

TFK70 replied:
И он не один здесь такой

TFK70 commented on practice/web-diary/ui/layout/src/Layout.tsx:
```diff
@@ -0,0 +1,18 @@
+
+import { LayoutType, RowType }                           from './types'
+
+export const Row = styled.div<RowType>(layout, space, flexbox, position)
```
Не видел чтобы ты его использовал где-то

TFK70 commented on practice/web-diary/ui/post-list/src/PostList.tsx:
```diff
@@ -0,0 +1,23 @@
+  const elements = data.map((item: any) => {
+    return (
+      <List key={item.id}>
+        <PostListItem> {item.post}</PostListItem>
```
Так обычно не делают
Передай item.post пропом в PostListItem и обработай его внутри айтема:
```javascript
<PostListItem content={item.post} />
```

gochicus committed:
refactor(web-diary): app structure refactored

affects: @app-launcher/app, @components/app-header, @components/post-add-form,
@components/post-list-item, @components/post-list, @ui/background

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/components/post-list-item/package.json:
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "@components/post-list-item",
```
Нет, post-list должен быть в ui
Есть идеи почему?

gochicus replied:
@TFK70 Нет, вообще не вижу логики с точки зрения зависимости от контекста. Но всё таки постлист или постлист айтэм?

gochicus committed:
refactor(web-diary): app structure refactor

affects: @app-launcher/app, @components/list-item, @ui/list

ISSUES CLOSED: #62

gochicus committed:
refactor(web-diary): app structure refactored

affects: @app-launcher/app, @ui/list

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/ui/list/src/ListItem.tsx:
```diff
@@ -1,16 +1,9 @@
 
 import { Button }      from '@ui/button'
 import { Box, Layout } from '@ui/layout'
+import { Label }       from '@ui/list'
```
Внутри пакета используй релятивные импорты
Такая запись подразумевает что `@ui/list` будет установлен как зависимость, но учитывая что ты уже работаешь в `@ui/list` это невозможно

TFK70 replied:
Да и если с точки зрения логики подумать: компонент ссылается сам на себя, выглядит странно

TFK70 commented on practice/web-diary/ui/list/src/Label.tsx:
```diff
@@ -0,0 +1,9 @@
+import styled from '@emotion/styled'
+
+export const Label = styled('span')(() => ({
```
Label точно должен быть описан в list?
Судя по тому что он стилизует span, Label можно заменить на компонент из `@ui/text`

gochicus committed:
refactor(web-diary): ccomponent refactored

affects: @ui/list

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/ui/post-status-filter/package.json:
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "@ui/filter",
```
1. Почему пакет пазывается `@ui/filter`, а компонент `PostStatusFilter`?
2. Ему точно место в ui?

gochicus replied:
@TFK70 с названием компонента да, упустил, а в ui, пожалуй, да. Это стилизованный див, которй оборачивает кннопки. В идеале, можно опустить фильтр и там же использовать лист, но тут не уверен, логически это другая часть приложения.

TFK70 replied:
> @TFK70 с названием компонента да, упустил, а в ui, пожалуй, да. Это стилизованный див, которй оборачивает кннопки. В идеале, можно опустить фильтр и там же использовать лист, но тут не уверен, логически это другая часть приложения.

В ui у тебя находятся скорее инструменты для создания компонентов, а не сами компоненты. Этот фильтр можно назвать инструментом?

gochicus replied:
@TFK70 Я перенесу его в компоненты,  но, по сути, структурно это тот же лист.

TFK70 replied:
@gochicus я не говорил же прямым текстом переносить его в компоненты, ты в праве сам решить что с ним делать, я просто указал что его текущее состояние вообще ни в какие ворота

gochicus committed:
refactor(web-diary): structure refactor

affects: @app-launcher/app, @components/filter, @components/form

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/components/filter/package.json:
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "@components/filter",
```
Ты не уловил суть
Компоненты из components - не универсальны, они привязаны к контексту и названия у них должны быть соответствующие, конкретизирующие места где они должны использоваться

gochicus replied:
@TFK70 Стоп, мы ведь не используем фрагменты сейчас? При чем тут они?  А, понятно, опечатка.

gochicus replied:
@TFK70  Если я кго в status-filter переименую, будет нормально? У компонента, как мне кажется, нет жесткой привязки к контексту.

TFK70 replied:
@gochicus вот тут ты по-хорошему и должен сам решить как для твоего приложения будет лучше
Ты его вынес в components потому что он реализует конкретный кусок логики
В данном случае он будет фильтровать тудушки и только. Если нам понадобится фильтровать юзеров - придется создавать новый компонент `@components/user-filter` который будет отвечать только за фильтрацию юзеров
Или ты можешь сделать `@ui/filter`, которому будет все-равно что фильтровать: юзеров или тудушки. Он будет делать это основываясь на переданных ему пропсах, а его назначение ты уже будешь определять сам в зависимости от контекста, как тебе удобно

gochicus replied:
@TFK70 тогда возвращаю его обратно в  ui, поскольку тут уловить адекватный контекст для привязки затруднительно. И по структуре он может принять любые инпуты и кнопки.

gochicus committed:
refactor(web-diary): structure change

affects: @app-launcher/app, @ui/filter

ISSUES CLOSED: #62

gochicus committed:
refactor(web-diary): structure refactor

affects: @app-launcher/app, @components/switcher-button

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/components/switcher-button/package.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "@ui/filter",
+  "name": "@components/switcher-button",
```
Почему button? 
1. Насколько я помню там было больше одной кнопоки
2. Название switcher уже подразумевает под собой компонент с каким-то количеством кнопок
3. Раз уж это находится в компонентах - где конкретика в названии? Что именно свитчер переключает?
4. И по-прежнему: ты уверен что это должно быть в components, а не в ui? Ты же можешь сделать универсальный компонент который будет создавать такие свитчеры, примерно как list

gochicus replied:
@TFK70 Именно потому, что switcher подразумевает группу toogle, я и указал button  - тип элементов, переключаемых switcher. Да, в компонентах с привязкой к контексту, в этой тудушке универсальный свитчер не нужен.

TFK70 replied:
@gochicus свитчер не подразумевает группу тоггл, это две разные вещи, хоть и похожие
Тоггл - это когда у тебя одна кнопка отвечает за переключение конкретного состояния (вкл/выкл, например)
Свитчер - когда у тебя есть панель из кнопок, каждая из которых отвечает за свое состояние. Это значит что за состояние вкл/выкл у тебя будут отвечать уже две соответствующие кнопки, а не одна

gochicus replied:
@TFK70 ну так тут и есть панель из кнопок, правда одна отвечает за удаление поста.

TFK70 replied:
@gochicus это не просто "панель из кнопок"
Если ты говоришь что это свитчер и он должен переключать состояния - значит там должны быть только кнопки, отвечающие за переключение состояния
Кнопка которая при нажатии удаляет пост - это не состояние, а вот например кнопка, которая помечает пост как прочитанный - это состояние

И я уже написал что означает название switcher-button. Ты называешь элемент в глобальном контексте, как из этого названия я должен понять какое состояние этот свитчер вообще переключает?

TFK70 replied:
@gochicus 
1. Не пренебрегай названиями
2. Не размывай зоны ответственности

gochicus committed:
refactor(web-diary): buttons panel extended, package name changed

affects: @app-launcher/app, @components/switcher-button-status, @ui/list

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/app-launcher/app/package.json:
```diff
@@ -9,7 +9,7 @@
     "@ui/list": "0.0.0",
     "@ui/text": "0.0.0",
-    "@components/switcher-button": "0.0.0",
+    "@components/switcher-button-status": "0.0.0",
```
Снова
Мне не стало понятнее какое состояние переключает свитчер. Я уже говорил что свитчер уже подразумевает в себе наличие кнопок, иначе это был бы не свитчер
Это слово status не прояснило абсолютно ничего. Я должен посмотреть на название пакета компонента и хотя бы примерно понять какую задачу он решает

TFK70 commented on practice/web-diary/app-launcher/app/src/App.tsx:
```diff
@@ -2,7 +2,7 @@ import React              from 'react'
 import { AppHeader }      from '@components/app-header'
 import { Form }           from '@components/form'
-import { SwitcherButton } from '@components/switcher-button'
+import { SwitcherButton } from '@components/switcher-button-status'
```
Название компонента соответственно должно меняться

gochicus committed:
refactor(web-diary): component name change

affects: @app-launcher/app, @components/switcher-button-post-status

ISSUES CLOSED: #62

TFK70 commented:
@gochicus опиши здесь почему ты назвал компонент именно так, за что отвечает каждая часть его названия

gochicus commented:
@TFK70 Сущность: switcher, элементы в свитчере - button,   что переключают кнопки  статус постов.

TFK70 commented:
@gochicus свитчера не может быть без кнопок внутри, зачем мне говорить что у машины есть колеса?

gochicus commented:
@TFK70 switcher-post-status

TFK70 commented:
@gochicus близко, но если забить switcher-post-status в переводчик то мы получим «статус поста коммутатора»
а если поменять на post-status-switcher то это уже будет «переключатель состояния сообщения»
про инглиш тоже забывать не стоит

TorinAsakura commented:
@TFK70 Запрашивая контекст мы должны задавать вопрос: что-где-делает

gochicus commented:
@TorinAsakura Вторая попытка

TorinAsakura commented:
@gochicus Очень хорошо. Теперь напиши то, что в прошлый раз не ушло

gochicus commented:
@TFK70 Так,  ещё вопрос, разве не важнее обозначить сущность в первую очередь?  Или главное, чтобы было понятно?

TFK70 commented:
@gochicus если придерживаться шаблона «что-где-делает» - то да, важнее, оставь switcher-post-status

TorinAsakura commented:
@TFK70 Это вы там фрагмент колдуете чтоли уже?

TFK70 commented:
@TorinAsakura почти
Саша решил не делать компонент switcher в ui и вынести его как конкретный кусок логики в components
вот и придумываем как назвать

gochicus committed:
refactor(web-diary): component name changed

affects: @app-launcher/app, @components/switcher-post-status

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/components/form/package.json:
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "@components/form",
```
С этим абсолютно аналогично
что - form
где - ???
делает - ???

gochicus committed:
refactor(web-diary): component name changed

affects: @app-launcher/app, @components/form-post-add

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/app-launcher/app/package.json:
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "@app-launcher/app",
```
Последний штрих
Зачем тебе app-launcher?
Сделай просто app и назови это @web-diary/app, так будет чище

TFK70 commented on practice/web-diary/app-launcher/app/package.json:
```diff
@@ -0,0 +1,20 @@
+    "@ui/list": "0.0.0",
+    "@ui/text": "0.0.0",
+    "@components/switcher-post-status": "0.0.0",
+    "@app-launcher/app": "0.0.0",
```
И у тебя здесь вроде как цикл зависимостей?

gochicus committed:
refactor(web-diary): package name changed

affects: @web-diary/app

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/app-launcher/app/package.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "@app-launcher/app",
+  "name": "@web-diary/app",
```
Хорошо, но зачем ты сейчас это в app-launcher держишь?
Бессмысленная вложенность, оставь просто папку app

TFK70 commented on practice/web-diary/package.json:
```diff
@@ -25,7 +25,7 @@
   },
   "dependencies": {
-    "@app-launcher/app": "0.0.0",
+    "@web-diary/app": "0.0.0",
```
ну вот зачем тебе здесь эта зависимость?
я уже 3 раз спрашиваю про нее

gochicus replied:
@TFK70 actl пишет, что нужна

TFK70 replied:
@gochicus ну правильно, ты же [здесь](https://github.com/Atlantis-Academy/frontend-study-gochicus/pull/65/files#diff-7752378a93292c3f352a708658c3bf3e44484a0db1b39f22aacb7a7300edbde2R5) зачем-то ссылаешься на @web-diary/app хотя уже работаешь в нем
Я уже писал что внутри пакетов нужно использовать релятивные импорты в таких случаях

gochicus committed:
refactor(web-diary): exessive dependency deleted

affects: @web-diary/app

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/app/index.tsx:
```diff
@@ -2,7 +2,7 @@ import React            from 'react'
 import { IntlProvider } from 'react-intl'
 
-import { App }          from '@web-diary/app'
+import App              from './src/App'
```
Будь внимательнее
У тебя в src есть index, импорт должен быть
```javascript
import App from './src'
```

TFK70 replied:
@gochicus  Точнее 
```javascript
import { App } from './src'
```
В индексах не должно быть дефолтных экспортов

gochicus committed:
refactor(web-diary): app import fixed

affects: @web-diary/app

ISSUES CLOSED: #62

TFK70 commented:
@gochicus messages из button. ты так и не убрал...

TFK70 commented on practice/web-diary/ui/button/messages/messages.ts:
```diff
@@ -0,0 +1,18 @@
+
+import { name }           from '../package.json'
+
+export default defineMessages({
```
Вот эти вот зачем вот они?

gochicus committed:
refactor(web-diary): messages from button directory deleted

affects: @web-diary/app

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/ui/text/src/index.tsx:
```diff
@@ -0,0 +1,3 @@
+import { Text } from './Text'
```
Я уже говорил про такую конструкцию
Либо звездочка, либо используй дефолтный экспорт в Text

gochicus replied:
@TFK70  Я тогда, к слову, интересовался а почему всё-таки звездочка лучше, чем эта конструкция?

TFK70 replied:
@gochicus я ж тогда и ответил

```javascript
import { Component } from './Component'

export { Component }
```

vs

```javascript
export * from './Component'
```

gochicus replied:
@TFK70 Эммм...это экономия места? Смысл не совсем понятен.

TFK70 replied:
@gochicus нуууу......дааа.....
кода в 2 раза меньше

TFK70 replied:
@gochicus тут вопрос не в экономии, незачем на паре байт экономить
звездочки лаконичнее и эквивалентны  конструкции import->export, этого достаточно чтобы их использовать

gochicus replied:
@TFK70  У меня звездочками не все компоненты корректно импортируются почему-то, так что где-то оставил старый импорт, сообщение нет такого дефолтного экспорта, хотя он есть из компонента.

gochicus committed:
refactor(web-diary): exports changed

affects: @components/switcher-post-status, @ui/background, @ui/button, @ui/input, @ui/list, @ui/text

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/components/app-header/src/index.tsx:
```diff
@@ -0,0 +1,3 @@
+import AppHeader from './AppHeader'
```
@gochicus в таких ситуациях так и должно быть, звездочки на дефолтных экспортах не работают

TFK70 commented on practice/web-diary/ui/list/src/index.tsx:
```diff
@@ -0,0 +1,5 @@
+import List         from './List'
+import { ListItem } from './ListItem'
```
Тебя тут ничего не смущает?

gochicus replied:
@TFK70  ты ж писал либо дефолтный экспорт либо звездочки

TFK70 replied:
@gochicus ну так блин а почему у тебя здесь и дефолтный экспорт, и именной экспорт
Раз уж ты везде звездочки делал то придерживайся одного стиля и убери дефолтный экспорт из List

gochicus committed:
refactor(web-diary): exports refactored

affects: @components/app-header, @components/form-post-add, @ui/list

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/components/app-header/src/AppHeader.tsx:
```diff
@@ -0,0 +1,18 @@
+
+import messages    from '../messages/messages'
+
+const AppHeader = () => {
```
export const тогда уж

TFK70 commented on practice/web-diary/components/app-header/src/AppHeader.tsx:
```diff
@@ -0,0 +1,18 @@
+  )
+}
+
+export { AppHeader }
```
чтобы этого не было

gochicus replied:
@TFK70 как должно быть тогда?  Просто  `export const AppHeader` ?

TFK70 replied:
@gochicus export const AppHeader = ...
Я ж напиал выше

gochicus committed:
refactor(web-diary): exports refactored

affects: @components/app-header, @components/form-post-add, @ui/list

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/ui/background/src/Background.tsx:
```diff
@@ -0,0 +1,8 @@
+export const Background = styled('div')(() => ({
+  margin: '50px 0 0 0',
+  padding: '0',
+  backgroundColor: '#eaf5fc',
```
Сашаааа
Мои самые первые комменты только и были про это: у твоего Background как минимум цвет-то должен поддаваться конфигурации

TFK70 replied:
Оставь его в defaultProps да и все, но для конфигурации открой

gochicus committed:
refactor(web-diary): components refactor

affects: @components/form-post-add, @components/switcher-post-status, @ui/background, @ui/list

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/components/form-post-add/src/FormPostAdd.tsx:
```diff
@@ -1,17 +1,20 @@
+import messages                                     from '../messages/messages'
 
-const StyledForm = styled('div')(() => ({
+const StyledForm = styled<any>('div')(border, color, layout, space, typography)
```
Вот тут-то этого быть и не должно
Если ты вынес форму в components - это значит что это кусок уже готовой логики, в данном случае форму конфигурировать/модифицировать и т.д. ни к чему т.к. она позиционируется как уже готовый компонент с реализованной логикой

И не пихай бездумно ВСЕ функции из styled-system везде подряд

TFK70 commented on practice/web-diary/components/switcher-post-status/src/SwitcherPostStatus.tsx:
```diff
@@ -1,16 +1,19 @@
+import messages                                     from '../messages/messages'
 
-const StyledSwitcher = styled('div')(() => ({
+const StyledSwitcher = styled<any>('div')(border, color, layout, space, typography)
```
Например тут
Зачем тебе typography? Это просто контейнер для итемов, зачем контейнеру для итемов применять стили для шрифта?

TFK70 replied:
Хотя заметил что он в components находится и здесь этого вообще быть не должно

TFK70 commented on practice/web-diary/ui/list/src/List.tsx:
```diff
@@ -1,13 +1,16 @@
+import { ListItem }                                 from './ListItem'
 
-const StyledList = styled('div')(() => ({
+const StyledList = styled<any>('div')(border, color, layout, space, typography)
```
Вот, здесь зато то же самое

Контейнер к которому зачем-то можно применять fontSize, fontFamily и прочее: зачем?
Здесь даже border сомнительный, оставь только нужное

TFK70 replied:
Если тебе нужно менять стили шрифта для лист итемов - вынеси typography в ListItem, это его зона ответственности

gochicus committed:
refactor(web-diary): components refactored

affects: @components/form-post-add, @components/switcher-post-status, @ui/background, @ui/list

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/ui/background/src/Background.tsx:
```diff
@@ -0,0 +1,11 @@
+import styled                                       from '@emotion/styled'
+import { border, color, layout, space, typography } from 'styled-system'
+
+export const Background = styled<any>('div')(border, color, layout, space, typography)
```
Еще раз: не торопись
Зачем тебе фону задавать бордер и стили для шрифта?
Пройдись еще раз по всему и проверь ui на такие моменты

gochicus replied:
@TFK70 Здесь прописаны CSS стили, что должны находиться в body,  поэтому шрифты там нужны. Я этим компонентом приложение обернул, возможно, название неудачное.

TFK70 replied:
@gochicus так у тебя бекграунд не должен отвечать за шрифты всего приложения, он отвечает за задний фон
Он не должен отвечать у тебя за шрифт вообще. За шрифт в инпуте отвечает typography в инпуте, за шрифт в баттоне тоже свои пропсы и т.д.

TFK70 replied:
@gochicus опиши стандартные стили шрифта в дефолт пропсах у всех чтобы везде одно и то же не писать

gochicus committed:
refactor(web-diary): defaultProps updated

affects: @ui/background, @ui/input, @ui/list

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/components/switcher-post-status/src/SwitcherPostStatus.tsx:
```diff
@@ -0,0 +1,27 @@
+  )
+}
+
+export default SwitcherPostStatus
```
Два экспорта?

gochicus replied:
@TFK70 изменения неудачно откатывал, есть ещё что-нибудь на исправление?

TFK70 replied:
@gochicus больше нет, ничего не заметил, исправь это и аппрувну
Но ты все-равно еще раз по проекту пройдись

gochicus replied:
@TFK70 Посмотрю уже утром, на свежую голову, сейчас уже глаз замылился.

gochicus committed:
refactor(web-diary): export refactor

affects: @components/switcher-post-status

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/ui/button/package.json:
```diff
@@ -0,0 +1,14 @@
+    "@emotion/styled": "^11.1.5",
+    "styled-system": "^5.1.5",
+    "@emotion/react": "^11.1.5",
+    "@ui/button": "0.0.0"
```
1. Ты нигде его не используешь
2. Это цикл зависимостей

TFK70 commented on practice/web-diary/ui/button/src/types.ts:
```diff
@@ -0,0 +1,4 @@
+export interface ButtonProps {
```
Их не нужно оставлять просто для галочки
Убери если их не используешь, от них никакого смысла нет

TFK70 commented on practice/web-diary/ui/button/src/Button.tsx:
```diff
@@ -0,0 +1,20 @@
+
+import { ButtonProps }                                from './types'
+
+export const Button = styled<any>('button')<ButtonProps>(border, color, layout, padding, typography)
```
Оставь только any, здесь 0 смысла от ButtonProps

TFK70 commented on practice/web-diary/ui/button/src/Button.tsx:
```diff
@@ -0,0 +1,20 @@
+import styled                                         from '@emotion/styled'
+import { border, color, layout, padding, typography } from 'styled-system'
```
Убери padding, его включает space из styled-system

gochicus committed:
refactor(web-diary): excessive dependency, unused interface - deleted

affects: @ui/button

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/ui/layout/src/types.ts:
```diff
@@ -0,0 +1,8 @@
+export interface LayoutType {
```
Лучше уж any написать вместо этого
Интерфейс все-равно не законченный, убери его тоже

TFK70 commented on practice/web-diary/ui/list/src/List.tsx:
```diff
@@ -0,0 +1,25 @@
+}
+
+export const List: any = ({ data }: any) => {
+  const elements = data.map((item: any) => {
```
Это не обязательно хранить в отдельной переменной, возвращай напрямую

gochicus committed:
refactor(web-diary): interface, const elements deleted

affects: @ui/layout, @ui/list

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/ui/list/src/List.tsx:
```diff
@@ -0,0 +1,24 @@
+
+export const List: any = ({ data }: any) => {
+  data.map((item: any) => {
+    return (
```
А зачем писать тело функции и return?
Сразу jsx и возвращай, этот огород ни к чему

gochicus replied:
@TFK70 не понял идею, как оно должно работать?

TFK70 replied:
@gochicus так же и должно
просто ты вместо
```javascript
arr.map(i => {
   return (<Jsx />)
})
```
 пишешь
```javascript
arr.map(i => <Jsx />)
```
азы синтаксиса

gochicus replied:
@TFK70  Такое имеешь в виду? 
 ```
data.map(item =>
      <StyledList key={item.id}>
        <ListItem content={item.post} />
      </StyledList>
```

TFK70 replied:
> @TFK70 Такое имеешь в виду?
> 
> ```
> data.map(item =>
>      <StyledList key={item.id}>
>        <ListItem content={item.post} />
>      </StyledList>
> )
> ```

да

gochicus replied:
@TFK70 Так не работает

TFK70 replied:
@gochicus если что ты скобку в конце потерял

gochicus replied:
@TFK70 она есть, пропустил при копировании

TFK70 replied:
@gochicus тогда конкретизируй что не работает, дай логи, предоставь код с точностью до символа

gochicus replied:
@TFK70  Похоже, просто баг был,  попытался воспроизвести - отрендерилось.

gochicus committed:
refactor(web-diary): component refactored

affects: @ui/list

ISSUES CLOSED: #62

TorinAsakura commented on practice/web-diary/app/src/App.tsx:
```diff
@@ -0,0 +1,34 @@
+    <Background>
+      <Layout flexDirection='column' alignItems='center' margin='0' maxWidth='800'>
+        <AppHeader />
+        <Layout margin='1 rem 0' justifyContent='space-between' alignItems='center'>
```
отступы считаем в статичных величинах

TorinAsakura commented on practice/web-diary/app/src/App.tsx:
```diff
@@ -0,0 +1,34 @@
+        <AppHeader />
+        <Layout margin='1 rem 0' justifyContent='space-between' alignItems='center'>
+          <Input placeholder='Поиск' />
+          <Box flexBasis='40px' />
```
ты наверно хотел сделать отступ? для отступов используем layout, box юзаем для стилизованных вещей

TorinAsakura commented on practice/web-diary/components/app-header/messages/messages.ts:
```diff
@@ -0,0 +1,10 @@
+export default defineMessages({
+  header: {
+    id: `${name}.header`,
+    defaultMessage: '5 записей, из них понравилось 0',
```
а если будет 10 записей и из них понравится 10? 5 и 10 не константы, это разные строки, разные объекты

TorinAsakura commented on practice/web-diary/components/app-header/package.json:
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "@components/app-header",
```
убери уже app-

TorinAsakura commented on practice/web-diary/components/form-post-add/src/FormPostAdd.tsx:
```diff
@@ -0,0 +1,31 @@
+    <StyledForm>
+      <Layout display='flex'>
+        <Box>
+          <Input placeholder='Ваша запись' />
```
а тут почему не проброшено через intl?

TorinAsakura commented on practice/web-diary/components/switcher-post-status/src/SwitcherPostStatus.tsx:
```diff
@@ -0,0 +1,25 @@
+import messages    from '../messages/messages'
+
+const StyledSwitcher = styled('div')(() => ({
+  margin: ' 1rem 0',
```
при попытке распарсить он увидит пробел в начале и ты словишь "что-то пошло не так"

TorinAsakura commented on practice/web-diary/ui/button/src/Button.tsx:
```diff
@@ -0,0 +1,18 @@
+  borderRadius: '4px',
+  cursor: 'pointer',
+  margin: '0 20px 0 0',
+  fontFamily: ' -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen"',
```
ебать-колотить держите меня крепче! Саша! Шойта за хуйня?

gochicus committed:
refactor(web-diary): any types deleted, components refactored

affects: @web-diary/app, @components/form-post-add, @components/header, @ui/background, @ui/button,
@ui/input, @ui/list, @ui/text

Использованы шрифты по умолчанию,
Андрей сказал, что они должны приходить из темы,
но на этом
проекте её пока не нужно

ISSUES CLOSED: #62

gochicus committed:
refactor(web-diary): excesive dependency deleted

affects: @web-diary/app

ISSUES CLOSED: #62

gochicus committed:
style(web-diary): lint changes

affects: @web-diary/app, @components/form-post-add, @components/header,
@components/switcher-post-status, @practice/web-diary, @ui/background, @ui/button, @ui/input,
@ui/list, @ui/text

ISSUES CLOSED: #62

gochicus committed:
style(web-diary): lint fix

affects: @web-diary/app, @components/form-post-add, @components/header,
@components/switcher-post-status, @practice/web-diary, @ui/background, @ui/button, @ui/input,
@ui/layout, @ui/list, @ui/text

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/components/form-post-add/messages/messages.ts:
```diff
@@ -0,0 +1,22 @@
+    id: `${name}.header`,
+    defaultMessage: 'Добавить',
+  },
+  placeholder: {
```
Слишком общее название
У тебя может быть несколько инпутов с разными плейсхолдерами на одной странице, конкретизируй

TFK70 commented on practice/web-diary/components/header/messages/messages.ts:
```diff
@@ -0,0 +1,10 @@
+import { name }           from '../package.json'
+
+export default defineMessages({
+  header: {
```
То же самое

TFK70 commented on practice/web-diary/components/header/package.json:
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "@components/header",
```
Конкретизируй название для хедера, напиши как минимум к чему он относится

TorinAsakura replied:
У него аппликуха одна, стало быть хедер относящийся к аппликухе. Тут нет смысла что-либо уточнять.

TorinAsakura commented on practice/web-diary/components/header/src/Header.tsx:
```diff
@@ -6,11 +6,11 @@ import { Text }    from '@ui/text'
   return (
     <Box justifyContent='space-between' alignItems='flex-end' display='flex'>
-      <Text fontSize='25px'>{intl.formatMessage(messages.header)}</Text>
+      <Text fontSize='25px'  >{post}{intl.formatMessage(messages.header)}{like}</Text>
```
Следи за вайтспейсами и чистотой синтаксиса

gochicus committed:
refactor(web-diary): messages changed, components refactored

affects: @components/form-post-add, @components/header, @practice/web-diary, @ui/list

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/components/form-post-add/messages/messages.ts:
```diff
@@ -15,7 +15,7 @@ export default defineMessages({
     defaultMessage: 'Добавить',
   },
-  placeholder: {
+  placeholderFormPostAdd: {
```
Тебе ни к чему в названиях месседжей привязывать их к контексту
Было бы неплохо называть их исходя из самого месседжа: yourPost было бы вполне себе ок
Иначе, если бы тебе пригодилось использовать это сообщение еще где-то помимо плейсхолдера, то сам понимаешь

gochicus committed:
refactor(web-diary): message name changed

affects: @components/form-post-add, @practice/web-diary

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/components/form-post-add/messages/messages.ts:
```diff
@@ -0,0 +1,22 @@
+    id: `${name}.header`,
+    defaultMessage: 'Добавить',
+  },
+  placeholderYourPost: {
```
Под контекстом я подразумевал слово placeholder

gochicus replied:
@TFK70  Я как-то потерял нить, здесь пробрасывается именно  текст плэйсхолдера, не содержимое поста или ещё что-то.  Я допускаю, может быть несколько разных плэйсхолдеров в компоненте и можно на каждый своё сообщение, с соответствующим названием. Но если просто YourPost, то тут, мне кажется, только путаница будет. Можешь прояснить?

TFK70 replied:
В чем именно путаница?
Путаница будет если тебе нужно будет использовать один и тот же текст и в плейсхолдере, и где-то помимо плейсхолдера, там придется создавать второй месседж с тем же содержимым, но с другим ключом

TFK70 replied:
И ты его назвал неверно, если у тебя в месседже "Добавить" - он должен называться add

yourPost - "Ваш пост"

TFK70 commented on practice/web-diary/components/form-post-add/messages/messages.ts:
```diff
@@ -0,0 +1,22 @@
+    defaultMessage: 'Добавить',
+  },
+  placeholderYourPost: {
+    id: `${name}.header`,
```
Так же и тут
Месседж "Ваша запись" почему-то имеет ключ header, при именовании месседжей отталкивайся от того, что в них

gochicus committed:
refactor(web-diary): message name changed

affects: @components/form-post-add, @practice/web-diary

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/components/form-post-add/messages/messages.ts:
```diff
@@ -0,0 +1,22 @@
+
+export default defineMessages({
+  info: {
+    id: `${name}.form-post-add`,
```
а что с id?
он должен соответствовать ключу месседжа, а слова должны разделяться андерскором

gochicus committed:
refactor(web-diary): messages id changed

affects: @components/form-post-add, @components/header, @practice/web-diary

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/components/header/messages/messages.ts:
```diff
@@ -4,11 +4,11 @@ import { name }           from '../package.json'
 export default defineMessages({
   headerPostCount: {
-    id: `${name}.header`,
+    id: `${name}.headerPostCount`,
```
Слова разделяются андерскором, я ж написал
header_post_count

TFK70 replied:
хотя тут тоже нужно изменить название аналогично тому как ты его поменял в месседже плейсхолдера, ты почему-то текст "записей," привязал к хедеру

TFK70 replied:
И пройдись по всем месседжам, сделай правки

TFK70 commented on practice/web-diary/components/form-post-add/src/FormPostAdd.tsx:
```diff
@@ -0,0 +1,31 @@
+
+import messages        from '../messages/messages'
+
+const StyledForm = styled('div')(() => ({
```
И обязательно ли было так делать?
У тебя же для таких махинаций существует Box

TFK70 commented on practice/web-diary/components/header/src/Header.tsx:
```diff
@@ -0,0 +1,21 @@
+export const Header = ({ post, like }) => {
+  const intl = useIntl()
+  return (
+    <Box justifyContent='space-between' alignItems='flex-end' display='flex'>
```
Я уже очень много раз говорил что display: flex уже есть в дефолт пропсах, зачем это здесь писать?

TFK70 commented on practice/web-diary/components/switcher-post-status/messages/messages.ts:
```diff
@@ -0,0 +1,18 @@
+
+export default defineMessages({
+  info: {
+    id: `${name}.switcher`,
```
Вот тут например тоже id везде одинаковые почему-то

TFK70 commented on practice/web-diary/ui/layout/src/Box.tsx:
```diff
@@ -0,0 +1,29 @@
+)
+
+Box.defaultProps = {
+  display: 'flex',
```
На всякий случай помечу где тот самый display:flex находится для наглядности

TFK70 commented on practice/web-diary/ui/list/src/List.tsx:
```diff
@@ -0,0 +1,21 @@
+  minWidth: '300px',
+}
+
+export const List: any = ({ data }: any) => {
```
Либо затипизируй нормально, либо ничего не пиши
В большинстве случаев будет достаточно типизировать пропсы и все

TFK70 commented on practice/web-diary/ui/list/src/List.tsx:
```diff
@@ -0,0 +1,21 @@
+
+const StyledList = styled('div')(layout, space)
+
+StyledList.defaultProps = {
```
И абсолютно то же самое
Что в этом блоке такого, что делает его листом? Это просто блок-обертка, а для их реализации есть Box

gochicus committed:
refactor(web-diary): components refactored, messages id changed

affects: @components/form-post-add, @components/header, @components/switcher-post-status,
@practice/web-diary, @ui/list

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/ui/background/package.json:
```diff
@@ -0,0 +1,13 @@
+    "react-intl": "^5.13.5",
+    "@emotion/styled": "^11.1.5",
+    "styled-system": "^5.1.5",
+    "@types/styled-system": "^5.1.11"
```
Где они должны находиться?)

gochicus replied:
@TFK70 Если ты имеешь ввиду тип стайлед систем, то ты ж и сам и говорил ставить их везде, где используются

TFK70 replied:
@gochicus да, но про @types я кажется тоже кое-что говорил
они нужны только при разработке

gochicus replied:
@TFK70 А, теперь понятно о чём ты

TFK70 commented on practice/web-diary/ui/button/package.json:
```diff
@@ -0,0 +1,14 @@
+    "@emotion/styled": "^11.1.5",
+    "styled-system": "^5.1.5",
+    "@emotion/react": "^11.1.5",
+    "@types/styled-system": "^5.1.11"
```
Где-где?)))

TFK70 commented on practice/web-diary/ui/list/src/List.tsx:
```diff
@@ -0,0 +1,13 @@
+import { Item as ListItem } from './Item'
+
+export const List = ({ data }: any) => {
+  return data.map(item => (
```
А я же вроде уже давал насчет этого комментарий
Зачем тебе полностью описывать тело функции с return когда можно сразу возвратить data.map(...) без всего этого

TFK70 commented on practice/web-diary/app/src/App.tsx:
```diff
@@ -0,0 +1,34 @@
+      <Layout flexDirection='column' alignItems='center' margin='0' maxWidth='800'>
+        <Header post={5} like={0} />
+        <Layout margin='10px 0' justifyContent='space-between' alignItems='center'>
+          <Input placeholder='Поиск' />
```
Почему тут не через intl?

TFK70 commented on practice/web-diary/app/src/App.tsx:
```diff
@@ -0,0 +1,34 @@
+  return (
+    <Background>
+      <Layout flexDirection='column' alignItems='center' margin='0' maxWidth='800'>
+        <Header post={5} like={0} />
```
Почему у тебя хедер принимает пропсы? Зачем?

TFK70 replied:
Опиши логику внутри него же

gochicus replied:
@TFK70 Дай подсказку в каком направлении думать, у меня в хэдере `<Text>  `обернутый в `<Box>`, ни тот ни другой к контексту привязывать не стоит.

gochicus committed:
refactor(web-diary): components refactored, messages id changed

affects: @web-diary/app, @components/form-post-add, @components/header,
@components/switcher-post-status, @practice/web-diary, @ui/background, @ui/button, @ui/input,
@ui/layout, @ui/list, @ui/text

ISSUES CLOSED: #62

TorinAsakura commented on practice/web-diary/app/package.json:
```diff
@@ -0,0 +1,20 @@
+    "@components/form-post-add": "0.0.0",
+    "@ui/list": "0.0.0",
+    "@ui/text": "0.0.0",
+    "@components/switcher-post-status": "0.0.0",
```
добавь по-человечески через yarn workspace

TFK70 commented on practice/web-diary/package.json:
```diff
@@ -0,0 +1,33 @@
+    "react": "^17.0.1",
+    "react-dom": "^17.0.2",
+    "react-intl": "^5.13.5",
+    "typescript": "^4.2.3"
```
Это дев зависимость, тебе тайпскрипт и типизация нужны только при разработке

TFK70 commented on practice/web-diary/components/header/package.json:
```diff
@@ -0,0 +1,12 @@
+    "react": "^17.0.1",
+    "@ui/layout": "0.0.0",
+    "react-intl": "^5.13.5",
+    "@ui/text": "0.0.0"
```
Тут тоже через yarn workspace добавь

gochicus committed:
refactor(web-diary): package.json files refactored

affects: @web-diary/app, @components/form-post-add, @components/header,
@components/switcher-post-status, @practice/web-diary

ISSUES CLOSED: #62

TFK70 commented:
@gochicus у тебя же чеки не прошли
Если что эти зависимости не надо было убирать - их надо было заново добавить через yarn workspace .. add
У тебя скорее всего он не находил локальные пакеты, чтобы нашел - прописывай руками версию пакета yarn add @ui/button@0.0.0

gochicus committed:
refactor(web-diary): packages refactored

affects: @web-diary/app, @components/form-post-add, @components/header,
@components/switcher-post-status, @practice/web-diary

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/package.json:
```diff
@@ -0,0 +1,33 @@
+  "scripts": {
+    "start": "webpack --mode development --watch",
+    "build": "webpack --mode production",
+    "lint": "actl lint"
```
А что он тут делает?

gochicus committed:
refactor(web-diary): package refactored

affects: @practice/web-diary

ISSUES CLOSED: #62

TorinAsakura commented on practice/web-diary/ui/list/src/List.tsx:
```diff
@@ -0,0 +1,12 @@
+
+export const List = ({ data }: any) =>
+  data.map(item => (
+    <Box marginTop='10px' padding='20px 35px 10px 35px' minWidth='300px' key={item.id}>
```
паддинги не должны использоваться для позиционирования родителей

gochicus committed:
refactor(web-diary): component refactored

affects: @practice/web-diary, @ui/list

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/ui/list/src/List.tsx:
```diff
@@ -6,7 +6,7 @@ import { Item as ListItem } from './Item'
 export const List = ({ data }: any) =>
   data.map(item => (
-    <Box marginTop='10px' padding='20px 35px 10px 35px' minWidth='300px' key={item.id}>
+    <Box marginTop='10px' minWidth='300px' key={item.id}>
```
жду переписанный кусок(ки)

gochicus committed:
refactor(web-diary): indents made using Layout component

affects: @web-diary/app, @components/form-post-add, @components/switcher-post-status,
@practice/web-diary, @ui/background, @ui/button, @ui/input, @ui/list

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/app/src/App.tsx:
```diff
@@ -15,20 +15,22 @@ const App = () => {
   return (
     <Background>
-      <Layout flexDirection='column' alignItems='center' margin='0' maxWidth='800'>
+      <Layout flexDirection='column' alignItems='center'>
```
а почему так-то?
у тебя блоки с flex-direction должны быть вынесены в разные компоненты Row и Column соответственно
это же вообще не похоже на тот пример который я показывал

gochicus replied:
@TFK70 ок,  поменяю, а в остальном как?

TFK70 replied:
@gochicus вроде норм, но после правок виднее будет

TFK70 commented on practice/web-diary/app/src/App.tsx:
```diff
@@ -0,0 +1,40 @@
+      <Layout flexDirection='column' alignItems='center'>
+        <Header />
+        <Layout flexBasis='20px' />
+        <Layout justifyContent='space-between' alignItems='center' minWidth='400px'>
```
Не пихай везде подряд Layout, в таких местах используй Box

gochicus replied:
@TFK70  Так Box же для стилизованных компонентов, а Layout для структуры?

TFK70 replied:
@gochicus да, но по-моему Layout лучше использовать для применения свойств которые влияют на разметку снаружи (типа align center), а Box для свойств которые более-менее заметно аффектят чилдренов

TFK70 replied:
@gochicus ну т.е. я в верстке Layout использую только когда мне нужно обернуть что-то в контейнер с минимумом свойств, который можно счесть скорее за формальность, чтобы верстка не полетела

для остального - Box

gochicus replied:
@TFK70   ` <Layout justifyContent='space-between' alignItems='center' minWidth='400px'>` Так здесь, вроде, и то и то, но в целом именно разметка. Возможно, если Layout с flexDirection поменять на Column, то избыток не так будет бросаться?

TFK70 replied:
@gochicus короче когда тебе что-то нужно обернуть старайся использовать Box, но когда чувствуешь что слишком жирно - меняй на Layout, я затрудняюсь объяснить где именно что использовать нужно, у меня это интуитивно)

TFK70 replied:
@gochicus я если что поправлять буду, со временем уловишь суть

gochicus committed:
refactor(web-diary): components refactored

affects: @web-diary/app, @components/form-post-add, @components/header,
@components/switcher-post-status, @practice/web-diary, @ui/layout, @ui/list, @ui/text

components Space, Column, Row added

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/ui/layout/src/Row.tsx:
```diff
@@ -0,0 +1,8 @@
+export const Row = styled.div(layout, space, flexbox, position, color, size)
+
+Row.defaultProps = {
+  display: 'flex',
```
а самое главное flexDirection: 'row' забыл)

gochicus replied:
@TFK70 У флекса display row по умолчанию же

TFK70 replied:
@gochicus да, но все-равно лучше добавь для наглядности

TFK70 commented on practice/web-diary/ui/text/src/Space.tsx:
```diff
@@ -0,0 +1,13 @@
+    space += space
+    rundownCounter -= 1
+  }
+  return <Text>{space} </Text>
```
пагади...
я же вижу что у тебя там пробел есть

TFK70 commented on practice/web-diary/ui/text/src/Space.tsx:
```diff
@@ -0,0 +1,13 @@
+import { Text } from './Text'
+
+export const Space = ({ count = 1 }) => {
+  let space = ''
```
зачем оно надо-то?
тут ни единого символа нет
то же что и 7+0+0+0+0

TFK70 commented on practice/web-diary/ui/text/src/Space.tsx:
```diff
@@ -0,0 +1,13 @@
+  let rundownCounter = count
+  while (rundownCounter >= 1) {
+    space += space
+    rundownCounter -= 1
```
если уж ты так делаешь - то делай через for чтобы лишние переменные не создавать

но вообще тут нужно использовать методы строк

TFK70 commented on practice/web-diary/ui/text/src/Space.tsx:
```diff
@@ -0,0 +1,13 @@
+import React    from 'react'
+
+import { Text } from './Text'
```
и тут нету большого смысла использовать текст, нам ни к чему применять какие-то стили к пробелу

TFK70 commented on practice/web-diary/components/header/src/Header.tsx:
```diff
@@ -1,24 +1,27 @@
+    <Row justifyContent='space-between' alignItems='flex-end'>
       <Text fontSize='25px'>
         {post}
+        <Space count={1} />
```
там же уже initialValue = 1, зачем здесь это  писать? просто < Space  />

TFK70 commented on practice/web-diary/ui/list/src/Item.tsx:
```diff
@@ -1,28 +1,28 @@
-    <Layout justifyContent='space-around'>
+    <Row justifyContent='space-around'>
       <Box minWidth='100px'>
         <Text>{content} </Text>
```
тут почему на space не заменил?

gochicus replied:
@TFK70  А здесь то зачем space? Тут контент поста в одно слово.

TFK70 replied:
@gochicus space как раз для того и вводился чтобы заменить такие пробелы
```javascript
<Text>{content}</Text>
<Space />
```

gochicus replied:
@TFK70 Если речь о пробеле между контентом и закрывающим Text, то здесь просто небрежность в коде, никакого пробела здесь не обозначается.

gochicus committed:
refactor(web-diary): components refactor

affects: @components/header, @practice/web-diary, @ui/layout, @ui/list, @ui/text

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/components/header/src/Header.tsx:
```diff
@@ -14,13 +14,13 @@ export const Header = () => {
-        {intl.formatMessage(messages.likeCount)}
-        <Space count={1} />
-        {like}
+        <span> {post} </span>
```
а это зачем??

TFK70 replied:
и если мне не изменяют мои глаза то там по бокам стоят пробелы

gochicus replied:
@TFK70 Возможно, я что-то упускаю, но просто так у меня корректно строки не склеиваются.

TFK70 replied:
@gochicus распиши в чем проблема тогда
ты же видишь что это полный бред и так точно быть не должно

но вообще основной вопрос был про то зачем ты тут Text на span поменял

TFK70 commented on practice/web-diary/ui/text/src/Space.tsx:
```diff
@@ -1,13 +1,6 @@
-    rundownCounter -= 1
-  }
-  return <Text>{space} </Text>
+  const space = ' '.repeat(count)
```
вот опять: зачем тебе создавать лишнюю переменную?
делай repeat прямо в jsx и возвращай сразу jsx

gochicus committed:
refactor(web-diary): component Space refactored

affects: @components/header, @practice/web-diary, @ui/text

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/ui/text/src/Space.tsx:
```diff
@@ -1,6 +1,5 @@
 export const Space = ({ count = 1 }) => {
-  const space = ' '.repeat(count)
-  return <span>{space}</span>
+  return <span>{' '.repeat(count)}</span>
```
почему вам так нравится этот return с телом функции писать?..
```javascript
const Text = ({children}) => {
   return <span>{children}</span>
}
```
vs
```javascript
const Text = ({children}) => <span>{children}</span>
```

TFK70 commented on practice/web-diary/components/header/src/Header.tsx:
```diff
@@ -14,13 +14,13 @@ export const Header = () => {
   return (
     <Row justifyContent='space-between' alignItems='flex-end'>
       <Text fontSize='25px'>
```
я бы не советовал сметать все это под один Text
тебе вполне может пригодиться в будущем сделать цвет шрифта у количества постов красным, например
вынеси все в отдельные Text и их уже раздели пробелами

gochicus committed:
refactor(web-diary): header refactored, unicode whitespace used

affects: @components/header, @practice/web-diary, @ui/text

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/components/header/src/Header.tsx:
```diff
@@ -13,15 +13,15 @@ export const Header = () => {
       <Text fontSize='25px'>
-        {post}
-        <Space />
         {intl.formatMessage(messages.postCount)}
```
стили шрифта применяются и к числам

gochicus replied:
@TFK70 Не понял суть проблемы.

TFK70 replied:
@gochicus сори, я чет даже несмотря на intl подумал что тут число находится...
почему ты postCount и likeCount то не разделил?

gochicus replied:
@TFK70  ` {intl.formatMessage(messages.postCount)}
                  <Space />
                 {intl.formatMessage(messages.likeCount)}`
Разделил же?  Если ты про разделение Text, то там одно предложение по смыслу, вряд ли окрашивать его в разные цвета будет хорошей идеей.

TFK70 replied:
@gochicus не, я про то почему ты в разные Text их не вынес

gochicus replied:
@TFK70 `записей, из них понравилось`  вот буквально эти две части предложения через запятую. Они в отдельных сообщениях на случай если понравившихся будет 0. Но вот стилизовать их по отдельности явно не нужно. 
К слову, поясни что ты имел ввиду в своем первом ревью про числа и шрифты.

TFK70 replied:
@gochicus если это имеет вид `x записей, из них понравилось y` - то `записей, из них понравилось` нужно вынести в один месседж

TFK70 replied:
> @TFK70 `записей, из них понравилось` вот буквально эти две части предложения через запятую. Они в отдельных сообщениях на случай если понравившихся будет 0. Но вот стилизовать их по отдельности явно не нужно.
> К слову, поясни что ты имел ввиду в своем первом ревью про числа и шрифты.

про числа и шрифты я сказал потому что думал что ты выводишь number переменную, типа:
```javascript
const x: number = 7
const Component = () => <p>{x}</p>
```
то есть так у тебя выведется 7 со всеми дефолтными стилями параграфа, + если ты припишешь какой-нибудь color='red' то 7 даже станет красной, ну и аналогично с остальными стилями шрифта, я хотел обратить внимание что при выводе чисел стили шрифта тоже важны

но, как я уже сказал, написал это по ошибке, так что все ок

gochicus committed:
refactor(web-diary): header messages refactored

affects: @components/header, @practice/web-diary

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/components/switcher-post-status/src/SwitcherPostStatus.tsx:
```diff
@@ -0,0 +1,17 @@
+  const intl = useIntl()
+  return (
+    <Row minWidth='180px' justifyContent='space-between'>
+      <Button color='black'> {intl.formatMessage(messages.all)}</Button>
```
пробеелы

gochicus replied:
@TFK70 Куда пробелы?

TFK70 replied:
@gochicus неужели опять `небрежность в коде`?)
в начале строки пробел есть, либо убери, либо оформи через space

TFK70 commented on practice/web-diary/ui/list/src/List.tsx:
```diff
@@ -0,0 +1,15 @@
+
+export const List = ({ data }: any) =>
+  data.map(item => (
+    <Column flexDirection='column'>
```
что-то тут есть лишнее

gochicus committed:
refactor(web-diary): flexDirection from components deleted

affects: @web-diary/app, @components/switcher-post-status, @practice/web-diary, @ui/list

ISSUES CLOSED: #62

TorinAsakura commented on practice/web-diary/ui/list/src/Item.tsx:
```diff
@@ -0,0 +1,28 @@
+      </Box>
+      <Box justifyContent='space-around' alignItems='center' minWidth='100px'>
+        <Button color='red'>
+          <i className='fa fa-star' />
```
откуда у тебя fontawesome тут взялся? переведи в компоненты все иконки и это можно сделать за счёт цикла

TorinAsakura reviewed (commented):
Поправить дребедень с иконками и 👍🏻

gochicus committed:
refactor(web-diary): icons added

affects: @web-diary/app, @practice/web-diary, @ui/icons, @ui/list

ISSUES CLOSED: #62

TFK70 commented on practice/web-diary/ui/list/src/Item.tsx:
```diff
@@ -0,0 +1,32 @@
+import { Box, Layout, Row } from '@ui/layout'
+import { Text }             from '@ui/text'
+
+import { HeartIcon }        from '../../icons'
```
ну импорты-то

gochicus committed:
refactor(web-diary): imports changed

affects: @practice/web-diary, @ui/list

ISSUES CLOSED: #62

TorinAsakura reviewed (approved):
Поздравляю дружище
